### ISSUE TITLE
Make UUID configuration optional in `os configure` 

### ISSUE THREAD

**Issue by brownjohnf** [resin-io/resin-cli#691]
As reported in the linked Front ticket, a user is attempting to configure an image, setting the wifi credentials and processor count, and then also preload an image. The user is able to configure via the Python SDK, but is unable to preload with the Python SDK. 

Conversely, the user is able to preload with the CLI, but is unable to configure (because the CLI configure command requires a UUID)

<img src="https://frontapp.com/assets/img/icons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_38e1)

**Comment by pimterry** [comment]
@shaunmulligan I'd like to do this, since it should be fairly easy and it sounds like it's a blocker for the linked customer. The only sensible way of handling it I can see is a breaking CLI change though, is that a concern? I'd like to go from:

```
resin os configure <image> <uuid>
resin os configure <image> <uuid> [deviceApiKey]
```

to

```
resin os configure <image> --app <appName>
resin os configure <image> --device <uuid>
resin os configure <image> --device <uuid> --deviceApiKey [deviceApiKey]
```

I think that fits in with our long-term design ideas for the CLI, and it matches the signature for `config generate`. Happy to go for it?

**Comment by shaunmulligan** [comment]
@pimterry , I think that makes sense. Are we marking breaking changes explicitly in the Change Log and also warn when something is deprecated?

**Comment by pimterry** [comment]
@shaunmulligan We don't have any deprecations that I know of right now. Yes though, all breaking changes should be marked as such in the changelog.

I was expecting to just break the CLI API and jump here to v7, rather than trying to support both formats but deprecate the older one. Is that what you're suggesting? We could try to support both in the short-term, but it's fairly awkward in the code (capitano particularly doesn't support this particularly well), so I'd rather not. Since it's a breaking release nobody will automatically upgrade, so we could just break it and people can stay on v6 until they have a minute to make the small change and move to the new format. Do you think we need a more complicated deprecation process here?

**Comment by pimterry** [comment]
Are you happy with the above? Should be quick and easy to do if so, I just need you to confirm.

**Comment by shaunmulligan** [comment]
@pimterry that sounds good. Will the multicontainer stuff also require any breaking changes or can we also just keep that in the v7.x.x line? I just don't want to have regular breaks :)

**Comment by pimterry** [comment]
I'm not sure yet, but it seems pretty likely that multicontainer will break things, and we'll definitely have a bunch of breakage as part of the CLI redesign too. You're right, that's going to be painful for some users, we should try to limit this a bit more.

Let's do this the hard way then: support the old command format and two new formats, print a deprecation message for the old format, and kill it off sometime later. Thanks :+1:

**Comment by shaunmulligan** [comment]
thanks @pimterry you're a champion

**Comment by pimterry** [comment]
Fixed in 6.10.0 - we still support the old format (now printing a warning), but also support both `--device <uuid>` and `--app <appname>` options. I'll update the customer on the front thread too.
